modules/react/collection/lib/useOverflowListModel.tsx

```diff
@@ -114,80 +116,101 @@ export const useOverflowListModel = createModelHook({
     select(data: Parameters<typeof model.events.select>[0]) {
       const {selectedIds} = model.selection.select(data.id, state);
       const ids = getHiddenIds(
-        containerWidthRef.current,
+        containerSizeRef.current,
         containerGap,
-        overflowTargetWidthRef.current,
-        itemWidthCacheRef.current,
+        overflowTargetSizeRef.current,
+        itemSizeCacheRef.current,
         selectedIds,
         config.items
       );
       model.events.select(data);
 
       setHiddenIds(ids);
     },
-    setContainerWidth(data: {width?: number}) {
-      containerWidthRef.current = data.width || 0;
-      setContainerWidth(data.width || 0);
-
+    setContainerSize(data: {width?: number; height?: number}) {
+      containerSizeRef.current =
+        model.state.orientation === 'horizontal' ? data.width || 0 : data.height || 0;
+      setContainerSize(containerSizeRef.current);
+      // setContainerGap(containerSizeRef.current);
       const ids = getHiddenIds(
-        containerWidthRef.current,
+        containerSizeRef.current,
         containerGap,
-        overflowTargetWidthRef.current,
-        itemWidthCacheRef.current,
+        overflowTargetSizeRef.current,
+        itemSizeCacheRef.current,
         state.selectedIds,
         config.items
       );
-
       setHiddenIds(ids);
     },
+    setContainerWidth(data: {width?: number}) {
```

Comment:
Add deprecated message here:

```suggestion
    /**
    * @deprecated Use `setContainerSize` instead and pass both `width` and `height`
    */
    setContainerWidth(data: {width?: number}) {
```